### PR TITLE
Remove exception handling; update docs

### PR DESCRIPTION
**Pull request by appukuttan-shailesh**
Implements following:
1) Part 1 of #17 (update exception handling)
2) Updates docstrings

To do:
1) Logging procedure

### CODE REVIEW

**Comment by appukuttan-shailesh** [comment]
@apdavison : Logging is currently being used only inside the `Container` class. Should we implement what we discussed in #17, or should we just do away with logging completely?

**Comment by apdavison** [comment]
I think logging will be helpful for our users, especially if directed to the terminal when using interactively. We should probably add log statements in the other classes, but it's not urgent.

File: hbp_archive.py [hunk 1]
```diff
@@ -892,20 +1003,39 @@ def get_container(self, name):
...
         return self._containers
 
     @property
     def container_names(self):
-        """Returns a list of container names"""
+        """Returns a list of container names
+
+        Returns
+        -------
+        list
```
Comment (appukuttan-shailesh) [inline hbp_archive.py]: @apdavison: At a few places, autodoc is automatically linking the return type `list` with `...#hbp_archive.PublicContainer.list`. Do you know how to suppress this, to just have plaintext?

Comment (appukuttan-shailesh) [reply hbp_archive.py]: Sample warning message during `make`:
> /home/shailesh/miniconda2/lib/python2.7/site-packages/hbp_archive.py:docstring of hbp_archive.Project.container_names:None: WARNING: more than one target found for cross-reference u'list': hbp_archive.PublicContainer.list, hbp_archive.Container.list

**Comment by appukuttan-shailesh** [comment]
@apdavison : could you check and merge this PR; I am not too confident of the logging part and would prefer to push it as a separate PR.

**Pull request merged**